### PR TITLE
Support apiserver-only nodes on GCE

### PR DESCRIPTION
**Pull request by justinsb**
Extending the existing AWS support to support GCE also.

### CODE REVIEW

File: pkg/apis/kops/validation/instancegroup.go [hunk 1]
```diff
@@ -187,8 +187,14 @@ func CrossValidateInstanceGroup(g *kops.InstanceGroup, cluster *kops.Cluster, cl
...
 
-	if g.Spec.Role == kops.InstanceGroupRoleAPIServer && kops.CloudProviderID(cluster.Spec.CloudProvider) != kops.CloudProviderAWS {
-		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "role"), "Apiserver role only supported on AWS"))
+	if g.Spec.Role == kops.InstanceGroupRoleAPIServer {
+		switch kops.CloudProviderID(cluster.Spec.CloudProvider) {
+		case kops.CloudProviderAWS, kops.CloudProviderGCE:
+			// Allowed
+
+		default:
+			allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "role"), "Apiserver role only supported on AWS"))
```
Comment (johngmyers) [inline pkg/apis/kops/validation/instancegroup.go:196]:
```suggestion
			allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "role"), "Apiserver role only supported on AWS and GCE"))
```

File: pkg/model/gcemodel/autoscalinggroup.go [hunk 2]
```diff
@@ -128,7 +128,7 @@ func (b *AutoscalingGroupModelBuilder) buildInstanceTemplate(c *fi.ModelBuilderC
 			}
 
 			switch ig.Spec.Role {
-			case kops.InstanceGroupRoleMaster:
+			case kops.InstanceGroupRoleMaster, kops.InstanceGroupRoleAPIServer:
```
Comment (johngmyers) [inline pkg/model/gcemodel/autoscalinggroup.go:131]: Do APIServer nodes really need DNS permissions?

Should't we tag them with InstanceGroupRoleAPIServer?

**Review by johngmyers — CHANGES_REQUESTED** [review]

File: pkg/model/gcemodel/autoscalinggroup.go (hunk 2 above)
Comment (justinsb) [reply pkg/model/gcemodel/autoscalinggroup.go:131]: Great points both - will fix!

Comment (justinsb) [reply pkg/model/gcemodel/autoscalinggroup.go]: Actually, the tag one is trickier... we share security groups on AWS as well, and this tag is a network tag, controlling the GCE equivalent of AWS security groups.

Comment (johngmyers) [reply pkg/model/gcemodel/autoscalinggroup.go]: Could use a comment then.

**Comment by justinsb** [comment]
/reopen

Not resolved

**Pull request closed without merging**